dataset/database.py

```diff
@@ -125,32 +124,22 @@ def begin(self):
 
         No data will be written until the transaction has been committed.
         """
-        if not hasattr(self.local, "tx"):
```

Comment:
wouldn't this remove support for nested transactions? as in  https://dataset.readthedocs.io/en/latest/quickstart.html#using-transactions